### PR TITLE
Funnel all writing through one function and add logging.

### PR DESCRIPTION
**Pull request by SirVer**

### CODE REVIEW

File: zvt/src/sequences.rs [hunk 1]
```diff
@@ -26,11 +26,28 @@ pub async fn read_packet_async(src: &mut Pin<&mut impl AsyncReadExt>) -> Result<
     buf.resize(start + len, 0);
     src.read_exact(&mut buf[start..]).await?;
 
-    debug!("Received {buf:?}");
+    if log_enabled!(Debug) {
```
Comment (dorezyuk) [inline zvt/src/sequences.rs:29]: i think this is redundant

File: zvt/src/sequences.rs [hunk 2]
```diff
@@ -91,7 +107,7 @@ where
             let bytes = read_packet_async(&mut src).await?;
             let packet = Self::Output::zvt_parse(&bytes)?;
             // Write the response.
-            src.write_all(&packets::Ack {}.zvt_serialize()).await?;
+            write_packet_async::<packets::Ack>(&mut src, &packets::Ack {}).await?;
```
Comment (dorezyuk) [inline zvt/src/sequences.rs:110]: i think you don't need to spell out the type here - it should be deduced from the argument

Comment (SirVer) [reply zvt/src/sequences.rs:110]: that is what I thought, but it does not compile without it.

File: zvt/src/sequences.rs (hunk 1 above)
Comment (SirVer) [reply zvt/src/sequences.rs:29]: no, calculating the hexstring is expensive and I think it is done even if the log statement is not executed. I was not able to verify this.

Comment (dorezyuk) [reply zvt/src/sequences.rs:29]: The logging macros check already check if the level is applicable before valuating the their arguments. The code 
```rs
fn main() {
    let world = "world";
    log::debug!("hello {}", world);
}
``

gets expanded to 
```rs
fn main() {
    let world = "world";
    {
        let lvl = ::log::Level::Debug;
        if lvl <= ::log::STATIC_MAX_LEVEL && lvl <= ::log::max_level() {
            ::log::__private_api::log(
                format_args!("hello {0}", world),
                lvl,
                &("fun", "fun", "src/main.rs"),
                5u32,
                ::log::__private_api::Option::None,
            );
        }
    };
}
```

Comment (SirVer) [reply zvt/src/sequences.rs]: oh, you are right, I checked with cargo expand now. Done.

**Comment by dorezyuk** [comment]
Are we closing this one?

**Comment by SirVer** [comment]
Yes, superseeded by #9.

**Pull request closed without merging**